src/core/api.proto

```diff
@@ -94,6 +94,55 @@ message InferRequestHeader
     //@@       for tensors with a non-fixed-size datatype (like STRING).
     //@@
     uint64 batch_byte_size = 3;
+
+    //@@      .. cpp:var:: message SharedMemoryElement
+    //@@
+    //@@         The Meta-data for the Input shared memory to read element from.
+    //@@
+    message SharedMemoryElement
```

Comment:
SharedMemoryElement and SharedMemoryIO should move to top-level so they can be shared for both input and output. Call them InferSharedMemoryElement and InferSharedMemory.